src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/ClientServices.cs

```diff
@@ -258,46 +483,53 @@ public class ClientServices : IClientHost, IEndpointServices, IEndpointDiscovery
         /// <returns></returns>
         internal IClientSession GetOrCreateSession(EndpointIdentifier id, bool persistent) {
             return _clients.GetOrAdd(id, k => new ClientSession(
-                CreateApplicationConfiguration(TimeSpan.FromMinutes(2), TimeSpan.FromMinutes(2)),
-                k.Endpoint, () => Certificate, _logger, NotifyStateChangeAsync, persistent,
+                _opcApplicationConfig, k.Endpoint, () => Certificate, _logger, NotifyStateChangeAsync, persistent,
                     _maxOpTimeout));
         }
 
         /// <summary>
         /// Create application configuration for client
         /// </summary>
         /// <returns></returns>
-        internal static ApplicationConfiguration CreateApplicationConfiguration(
+        internal ApplicationConfiguration CreateApplicationConfiguration(
             TimeSpan operationTimeout, TimeSpan sessionTimeout) {
+
+            // mitigation for bug in .NET Core 2.1
```

Comment:
Not sure if we call it bug - I think it is by design on windows, and platform specific.   But if it is a bug would be great to add the issue number here so we can track.

Reply:
Sure, we will have to keep this in mind, we will have issues in the future with it. I called it a bug because it has something to do with loading the .pfx from the folder storage. The certificate is correctly build and it can be imported manually in the windows certs manager. Problematically this fails.